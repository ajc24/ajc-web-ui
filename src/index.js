/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
