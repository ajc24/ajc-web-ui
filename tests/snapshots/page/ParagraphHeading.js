import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { ParagraphHeading } from '../../../src';

describe('ParagraphHeading', () => {
  describe('Default props and rendering - Component with left aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphHeading>
            Test paragraph heading.
          </ParagraphHeading>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with center aligned text and reduced upper spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphHeading alignment="center" additionalUpperSpacing={false}>
            Test paragraph heading.
          </ParagraphHeading>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with right aligned text', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ParagraphHeading alignment="right">
            Test paragraph heading.
          </ParagraphHeading>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
