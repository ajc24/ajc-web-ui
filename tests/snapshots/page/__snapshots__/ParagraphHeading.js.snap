// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParagraphHeading Default props and rendering - Component with left aligned text verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left ajc-spacing-top-large"
>
  <h2
    className="ajc-font-paragraph-heading"
  >
    Test paragraph heading.
  </h2>
</div>
`;

exports[`ParagraphHeading Transferred props and rendering - Component with center aligned text and reduced upper spacing verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-center"
>
  <h2
    className="ajc-font-paragraph-heading"
  >
    Test paragraph heading.
  </h2>
</div>
`;

exports[`ParagraphHeading Transferred props and rendering - Component with right aligned text verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-right ajc-spacing-top-large"
>
  <h2
    className="ajc-font-paragraph-heading"
  >
    Test paragraph heading.
  </h2>
</div>
`;
